scene/main/window.cpp

```diff
@@ -1250,16 +1242,10 @@ Rect2i Window::get_usable_parent_rect() const {
 }
 
 void Window::add_child_notify(Node *p_child) {
-	Control *child_c = Object::cast_to<Control>(p_child);
-
-	if (child_c && child_c->data.theme.is_null() && (theme_owner || theme_owner_window)) {
-		Control::_propagate_theme_changed(child_c, theme_owner, theme_owner_window); //need to propagate here, since many controls may require setting up stuff
-	}
-
-	Window *child_w = Object::cast_to<Window>(p_child);
-
-	if (child_w && child_w->theme.is_null() && (theme_owner || theme_owner_window)) {
-		Control::_propagate_theme_changed(child_w, theme_owner, theme_owner_window); //need to propagate here, since many controls may require setting up stuff
+	// We propagate when this node uses a custom theme, so it can pass it on to its children.
+	if (theme_owner || theme_owner_window) {
+		// `p_notify` is false here as `NOTIFICATION_THEME_CHANGED` will be handled by `NOTIFICATION_ENTER_TREE`.
+		Control::_propagate_theme_changed(this, theme_owner, theme_owner_window, false, true);
```

Comment:
Here "this" seems to be "p_child", `Control::_propagate_theme_changed` may switch `p_assign` if the same node is set multiple times, so propagation may be interrupted.

Reply:
@Rindbee I think you're totally right, I missed that. I think I was using `this` because originally this code actually was using `NOTIFICATION_PARENTED`, but I realized I'd have to change less code and I thought it'd have the same effect if I just used `add_child_notify` instead, since the only time theme items need to be propagated from a parent node is when that parent node is a control/window.

Reply:
I fixed this `this` issue in https://github.com/godotengine/godot/pull/65250 btw.